tests/jira/OnPremOAuth.ts

```diff
@@ -0,0 +1,247 @@
+            expect(normalizeRequestParams(new Map([c, a]))).to.equal("aachen=dom&cologne=maus");
+        });
+        it("url-encodes values", () => {
+            expect(normalizeRequestParams(new Map([d]))).to.equal("dortmund=Universit%C3%A4t");
+        });
+        it("handles different amounts of pairs", () => {
+            expect(normalizeRequestParams(new Map([a]))).to.equal("aachen=dom");
+            expect(normalizeRequestParams(new Map([a, b]))).to.equal("aachen=dom&berlin=fernsehturm");
+            expect(normalizeRequestParams(
+                new Map([a, b, c]))
+            ).to.equal("aachen=dom&berlin=fernsehturm&cologne=maus");
+            expect(normalizeRequestParams(
+                new Map([a, b, c, d]))
+            ).to.equal("aachen=dom&berlin=fernsehturm&cologne=maus&dortmund=Universit%C3%A4t");
+        });
+    });
+    describe("sortRequestParams", () => {
+        it("correctly sorts request params", () => {
+            const a: [string, string] = ["aachen", "dom"];
+            const b: [string, string] = ["aachen", "rathaus"];
+            const c: [string, string] = ["berlin", "fernsehturm"];
+            const d: [string, string] = ["berlin", "rotes rathaus"];
+            expect(sortRequestParams([a, b, c, d])).to.deep.equal([a, b, c, d]);
+            expect(sortRequestParams([d, c, b, a])).to.deep.equal([a, b, c, d]);
+            expect(sortRequestParams([c, b, a, d])).to.deep.equal([a, b, c, d]);
+            expect(sortRequestParams([a, b, a, d])).to.deep.equal([a, a, b, d]);
+        });
+    });
+    describe("usesCustomPort", () => {
+        it("returns true for a non-default HTTP port", () => {
+            expect(usesCustomPort(new URL('http://example.com:8008'))).to.be.true;
+            expect(usesCustomPort(new URL('http://example.com:8080'))).to.be.true;
+        });
+        it("returns true for a non-default HTTPS port", () => {
+            expect(usesCustomPort(new URL('https://example.com:1234'))).to.be.true;
+            expect(usesCustomPort(new URL('https://example.com:8443'))).to.be.true;
+        });
+        it("returns false when no port is specified", () => {
+            expect(usesCustomPort(new URL('http://example.com'))).to.be.false;
+        });
+        it("returns false for default ports", () => {
+            expect(usesCustomPort(new URL('http://example.com:80'))).to.be.false;
+            expect(usesCustomPort(new URL('https://example.com:443'))).to.be.false;
+        });
+    });
+});
+
+describe("JiraOnPremOAuth helper functions", () => {
+    const config: BridgeConfigJiraOnPremOAuth = {
+        consumerKey: "abc",
+        privateKey: "abc.pem",
+        // eslint-disable-next-line camelcase
+        redirect_uri: "https://redirect",
+    };
+    // This private key is ONLY used for testing. It's safe to be published.
+    const privateKey = `-----BEGIN PRIVATE KEY-----
```

Comment:
GitHub key scanner report coming in 10... 9... 8... 7...

As mentioned in the comment, I generated this key just for testing.